crates/iceberg-catalog/src/implementations/postgres/tabular/view/load.rs

```diff
@@ -0,0 +1,341 @@
+            let timestamp = r.timestamp;
+            let dni: Option<Uuid> = r.default_namespace_id;
+            let namespace_name = Self::get_namespace_ident_with_empty_support(conn, dni).await?;
+            let view_version_unique_id = r.view_version_unique_id;
+            let version_id = r.version_id;
+
+            let builder = ViewVersion::builder()
+                .with_timestamp_ms(timestamp.timestamp_millis())
+                .with_version_id(version_id)
+                .with_default_namespace(namespace_name)
+                .with_default_catalog(r.default_catalog)
+                .with_schema_id(r.schema_id)
+                .with_summary(Self::fetch_metadata_summary(conn, view_version_unique_id).await?)
+                .with_representations(
+                    Self::fetch_view_representations(conn, view_version_unique_id).await?,
+                )
+                .build();
+
+            versions.insert(version_id, Arc::new(builder));
+        }
+        Ok(versions)
+    }
+
+    async fn fetch_view_representations(
+        conn: &mut PgConnection,
+        view_version_unique_id: Uuid,
+    ) -> Result<Vec<ViewRepresentation>> {
+        Ok(sqlx::query!(
+            r#"
+                SELECT typ as "typ: ViewRepresentationType", sql, dialect
+                FROM view_representation
+                WHERE view_version_unique_id = $1
+                "#,
+            view_version_unique_id
+        )
+        .fetch_all(&mut *conn)
+        .await
+        .map_err(|e| {
+            let message = "Error fetching view representations".to_string();
+            tracing::warn!("{}", message);
+            e.into_error_model(message)
+        })?
+        .into_iter()
+        .map(|r| match r.typ {
+            ViewRepresentationType::Sql => {
+                ViewRepresentation::SqlViewRepresentation(SqlViewRepresentation {
+                    sql: r.sql,
+                    dialect: r.dialect,
+                })
+            }
+        })
+        .collect())
+    }
+
+    async fn fetch_metadata_summary(
+        conn: &mut PgConnection,
+        view_version_unique_id: Uuid,
+    ) -> Result<HashMap<String, String>> {
+        Ok(sqlx::query!(
+            r#"
+                SELECT key, value
+                FROM metadata_summary
+                WHERE view_version_unique_id = $1
+                "#,
+            view_version_unique_id
+        )
+        .fetch_all(&mut *conn)
+        .await
+        .map_err(|e| {
+            let message = "Error fetching metadata summary".to_string();
+            tracing::warn!("{}", message);
+            e.into_error_model(message)
+        })?
+        .into_iter()
+        .map(|r| (r.key, r.value))
+        .collect())
+    }
+
+    // this is a horrible function, and it's only here because NamespaceIdent doesn't allow constructing
+    // from empty vecs which is something that spark is handing to us.
+    async fn get_namespace_ident_with_empty_support(
```

Comment:
I asked over at apache on how to handle it. If you have an example Metadata file from spark, it would be great to add it to the PR at with my question at https://github.com/apache/iceberg-rust/pull/331#issuecomment-2204539127